### ISSUE TITLE
Joomla 4 HtmlHelper::_('calendar', ...) throws PHP notice 

### ISSUE THREAD

**Issue by nikosdion** [joomla/joomla-cms#29340]
### What needs to be fixed

Let's say you want to display a date picker (calendar) in Joomla 4.0.0.b1. As per usual you do:
```php
echo \Joomla\CMS\HTML\HTMLHelper::('calendar', $myDate, 'fieldName', 'fieldID');
```

You'd expect to see a Calendar field. However, if error reporting is set to Maximum you are met with a nasty PHP Notice "Notice: Undefined variable: dataAttribute in ...../layouts/joomla/form/field/calendar.php on line 117" This breaks the display and the calendar field.

### Why this should be fixed

Because it breaks all of my software where a date needs to be selected and I don't wanna have to implement my own date picker.

### How would you fix it

Change this https://github.com/joomla/joomla-cms/blob/4.0-dev/layouts/joomla/form/field/calendar.php#L117

to that:
```php
<?php echo $dataAttribute ?? ''; ?>
```

### Side Effects expected

None.

**Comment by brianteeman** [comment]
For reference this was merged without tests #29252

**Comment by nikosdion** [comment]
@brianteeman Thanks! I had tested this on Thursday and I was so confused as to why it suddenly broken over the weekend. Sucks that I'm using it with all of my software which I plan to release officially J4 compatible versions for in the next 10 days :/ At least if it's fixed it won't appear that I'm doing something wrong.

**Comment by brianteeman** [comment]
Well you know the quick way to get it fixed ;)

I don't understand that code so I would be working blind

**Comment by nikosdion** [comment]
Done :)

**Comment by alikon** [comment]
closed as we have #29347
